drivers/digital-io/max22196/iio_max22196.c

```diff
@@ -0,0 +1,682 @@
+
+	return sprintf(buf, "%s", max22196_current_avail[val]);
+}
+
+/**
+ * @brief Write the current attribute for a specific channel
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22196_iio_write_current(void *dev, char *buf, uint32_t len,
+				      const struct iio_ch_info *channel,
+				      intptr_t priv)
+{
+	struct max22196_iio_desc *desc = dev;
+	uint32_t i;
+	int ch;
+
+	ch = channel->address;
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(max22196_current_avail); i++)
+		if (!strcmp(buf, max22196_current_avail[i]))
+			break;
+
+	if (i == NO_OS_ARRAY_SIZE(max22196_current_avail))
+		return -EINVAL;
+
+	i = no_os_field_prep(MAX22196_CURR_MASK, i);
+
+	return max22196_reg_update(desc->max22196_desc, MAX22196_CFG_REG(ch),
+				   MAX22196_CURR_MASK, i);
+}
+
+/**
+ * @brief Read the current available attribute for a specific channel
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22196_iio_read_current_available(void *dev, char *buf,
+		uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	int length = 0;
+	int i;
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(max22196_current_avail); i++)
+		length += sprintf(buf + length, "%s ", max22196_current_avail[i]);
+
+	return length;
+}
+
+/**
+ * @brief Register read wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22196_iio_reg_read(struct max22196_iio_desc *dev, uint32_t reg,
+				 uint32_t *readval)
+{
+	return max22196_reg_read(dev->max22196_desc, reg, readval);
+}
+
+/**
+ * @brief Register write wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22196_iio_reg_write(struct max22196_iio_desc *dev, uint32_t reg,
+				  uint32_t writeval)
+{
+	return max22196_reg_write(dev->max22196_desc, reg, writeval);
+}
+
+/**
+ * @brief Setup the channels for the IIO device
+ * @param desc - MAX22196 IIO device descriptor
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+static int max22196_iio_setup_channels(struct max22196_iio_desc *desc)
+{
+	struct iio_channel *max22196_iio_channels;
+	uint32_t i;
+
+	max22196_iio_channels = no_os_calloc(MAX22196_CHANNELS,
+					     sizeof(*max22196_iio_channels));
```

Comment:
there is no need to allocate these channels dynamically seeing how they do not change function or number or anything else